pkg/config/validation_test.go

```diff
@@ -668,3 +668,75 @@ func TestValidateExportKubeConfig(t *testing.T) {
 		})
 	}
 }
+
+const patchesPath = "spec.containers[*].name"
+
+func TestSimpleValidateAllSyncPatches(t *testing.T) {
+	sync := config.Sync{
+		ToHost: config.SyncToHost{
+			Pods: config.SyncPods{
+				Patches: []config.TranslatePatch{
+					{
+						Path:              patchesPath,
+						Expression:        "\"my-prefix-\"+value",
+						ReverseExpression: "value.slice(\"my-prefix\".length)",
+					},
+				},
+			},
+		},
+	}
+	if err := ValidateAllSyncPatches(sync); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestMultipleValidateAllSyncPatches(t *testing.T) {
+	sync := config.Sync{
+		ToHost: config.SyncToHost{
+			Pods: config.SyncPods{
+				Patches: []config.TranslatePatch{
+					{
+						Path:              patchesPath,
+						Expression:        "\"my-prefix-\"+value",
+						ReverseExpression: "value.slice(\"my-prefix\".length)",
+					},
+				},
+			},
+			Endpoints: config.EnableSwitchWithPatches{
+				Patches: []config.TranslatePatch{
+					{
+						Path:              patchesPath,
+						Expression:        "\"my-prefix-\"+value",
+						ReverseExpression: "value.slice(\"my-prefix\".length)",
+					},
+				},
+			},
+		},
+	}
+	if err := ValidateAllSyncPatches(sync); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestValidateAllSyncPatchesNoDuplicatedPaths(t *testing.T) {
+	sync := config.Sync{
+		ToHost: config.SyncToHost{
+			Pods: config.SyncPods{
+				Patches: []config.TranslatePatch{
+					{
+						Path:       patchesPath,
+						Expression: "\"my-prefix-\"+value",
+					},
+					{
+						Path:              patchesPath,
+						ReverseExpression: "value.slice(\"my-prefix\".length)",
+					},
+				},
+			},
+		},
+	}
+	err := ValidateAllSyncPatches(sync)
```

Comment:
minor: Have you thought about consolidating this 3 `Test*` function into one and using `t.Run("<test-name>", func()` to run them as subtests? I think we use this approach in most of the unit tests we have (example: https://github.com/loft-sh/vcluster/blob/main/pkg/config/validation_test.go#L141-L145).
 I guess in this case, the test case struct could contain fields `config.Sync` as an input and `expectedErr error` which would be used in assertion

Reply:
Sure! I'll do that!

Reply:
PTAL